src/Microsoft.AspNetCore.Http/Extensions/HttpRequestRewindUtility.cs

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http.Internal;
+
+namespace Microsoft.AspNetCore.Http.Extensions
```

Comment:
This is a very odd namespace. It's even more odd that there's 1 type in it. And even if we add a 2nd type, it's still quite odd. Thoughts?

Reply:
Won't be in this namespace because I'll be changing to extension methods in the next iteration. Think that addresses all but your class-level doc comments concern.

Hmm, do we document `[EditorBrowsable(EditorBrowsableState.Never)]` classes?

Reply:
Why would it be browsable never? We shouldn't need to apply that to anything...

Reply:
But yes that otherwise sounds good thanks!